coderd/rbac/authz.go

```diff
@@ -870,17 +876,54 @@ func GetAuthzCheckRecorder(ctx context.Context) (*AuthzCheckRecorder, bool) {
 }
 
 // String serializes all of the checks recorded, using the following syntax:
+// name1=result1; name2=result2; ...
 func (r *AuthzCheckRecorder) String() string {
 	r.lock.Lock()
 	defer r.lock.Unlock()
 
-	if len(r.checks) == 0 {
+	// If we haven't written anything yet.
+	if r.writeIdx == 0 && !r.wrapped {
 		return "nil"
 	}
 
-	checks := make([]string, 0, len(r.checks))
-	for _, check := range r.checks {
-		checks = append(checks, fmt.Sprintf("%v=%v", check.name, check.result))
+	// Determine how many valid entries we have.
+	var count int
+	if r.wrapped {
+		count = MaxRecordedChecks
+	} else {
+		count = r.writeIdx
+	}
+
+	checks := make([]string, 0, count)
+
+	if r.wrapped {
```

Comment:
We should also include a warning in the output if we ended up wrapping

Reply:
This is obviated now that I'm using `container/ring`.